chore: updated PR template (#91)
zama-ai/kms: .github/PULL_REQUEST_TEMPLATE.md

- [ ] I have made sunshine tests for all new `pub` methods.
- [ ] Code comments are in place for public methods along with tricky or non-obvious code segments.
- [ ] I have performed a self-review of my code
- [ ] Any unfinished business is documented with a TODO and a reference to an issue describing what needs to be fixed.
- [ ] I have only used `unwrap`, `expect` or `panic!` tests or in situations where it would imply that there is a bug in the code, and I have documented this.
- [ ] My PR is _not_ updating _any_ dependencies (i.e. no changes to `cargo.lock`). Or if it is, then it _only_ contains the dependency updates and any changes needed to fix compilation and tests.
- [ ] My changes do not affect the architecture of the protocol. Or if they do these steps must be taken:
    - [ ] A parallel PR or issue has been open in the [tech-spec repo](https://github.com/zama-ai/tech-spec) (add the link here).
- [ ] My PR does not contain any breaking changes to the configuration and deployment files. Or if it does then these steps must be taken:
    - [ ] My PR is labeled with `devops`.

>     - [ ] I have pinged the infra team on Slack (in the MPC channel) and I have updated the intermediate release notes to document this change. 

Author: We don't have any right now. @dd23 should we add [this one](https://github.com/zama-ai/planning-blockchain/blob/main/.github/ISSUE_TEMPLATE/mpc-delivery.md) to the MPC repo s.t. we can incrementally update it?

>     - [ ] I have pinged the infra team on Slack (in the MPC channel) and I have updated the intermediate release notes to document this change. 

Reviewer: I think that particular template might be outdated, but I guess what's relevant is the changelog (of breaking changes). I don't think we have to commit that as a file, but we could collect it in the repo wiki or on Notion.

>     - [ ] I have pinged the infra team on Slack (in the MPC channel) and I have updated the intermediate release notes to document this change. 

Author: @dd23 what is wrong with collecting this in file? We used to do that at some point.
However, to avoid potential issues I must admit I think it better to just GH to keep track of this. 
Maybe I should just remove this part of the sentence? Is that ok @dd23 ?

>     - [ ] I have pinged the infra team on Slack (in the MPC channel) and I have updated the intermediate release notes to document this change. 

Author: Corrected in f50b25b9bcd6aafca02641f96e89d3442e31eacc

    - [ ] The Gateway and Connector teams have been notified about this change. 
- [ ] I have not changed existing `versionized` structs, nor added new `versionized` structs. Or if I have, these steps must be taken:
    - [ ] The backwards compatibility tests have been updated and/or new tests covering the changes have been added. 
- [ ] My PR does not contain changes to the critical business logic or cryptographic code. Or if it does then these steps must be taken:
    - [ ] At least two people must be assigned as reviewers (and eventually approve!) the PR.
- [ ] I have not added new structs or modified struct to contain private or key data. Or if so then these steps must be taken:
    - [ ] The `zeroize` and `ZeroizeOnDrop` traits have been implemented to clean up private data.
- [ ] I have not added data to the public storage. Or if I have, then these steps must be taken:
    - [ ] I have ensured that the data does _not_ need to be trusted. I.e. it can be validated through a signature or the existence of a digest in the private storage. 

> ### Checklist for dependency updates

Author: That is a great question!  I am actually not sure. I think we kinda have to accept secondary dependency updates. I think the alternative is to manually update the `cargo.lock` file, which I am not sure is the best idea. 
@dd23 WDYT?

> ### Checklist for dependency updates

Reviewer 2: sorry forgot to paste the example https://github.com/zama-ai/kms/pull/83/files

> ### Checklist for dependency updates

Author: Yes, I agree. So we assume automatic secondary updates of `cargo.lock` in connection with other updates is acceptable. 
Updated the text in 5165cfb951dabdd9ca4654664d7fe4f5f23eb417

## Describe your changes
<!-- Please explain the changes you made -->

<!--
Please, make sure:
- you have read the contributing guidelines:

>   https://github.com/zama-ai/kms/blob/main/docs/CONTRIBUTING.md

Author: Added an issue to take care of that here https://github.com/zama-ai/kms-internal/issues/2735

- [ ] My PR title follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), e.g. "chore: made key gen consistent with tfhe-rs 1.4".
- [ ] I have made sunshine tests for all new `pub` methods.
- [ ] Code comments are in place for public method along with tricky or non-obvious code segments.
- [ ] I have performed a self-review of my code

> - [ ] I have not used `unwrap`. Or if I have it is only in tests. 

Reviewer: ```suggestion
- [ ] I have not used `unwrap()` or `expect()`, except in clearly documented places or in tests.
```

Author: I don't think `unwrap` should be used in production code. `expect` is fine, but `unwrap` does not force documentation unlike `expect`

Author: ok fair enough. Updated in 87574926fd21ec383c5d51c2cc8361fe01f2d7e2

- [ ] My changes do not affect the architecture of the protocol. Or if they do these steps must be taken:
    - [ ] A parallel PR or issue has been open in the [tech-spec repo](https://github.com/zama-ai/tech-spec) (add the link here).
- [ ] My PR does not contain any breaking changes to the configuration and deployment files. Or if it does then these steps must be taken:
    - [ ] My PR is labeled with `devops`.
    - [ ] I have pinged the infra team on Slack (in the MPC channel) and I have updated the intermediate release notes to document this change. 
    - [ ] I have put a devops person on the PR as reviewer.
- [ ] My PR does not contain breaking changes to the gRPC interface or data serialized into data in the service gRPC interface. In particular there are no changes to the `extraData` fields. Or if it does the following steps have been taken:
    - [ ] The PR is marked using `!` in accordance with conventional commits. E.g. `chore!: changed decryption format according to Q3 release`. 

>     - [ ] The gateway teams has been notified about this change. 

Reviewer: should we rather mention Connector here?

- [ ] I have not changes existing `versionized` structs, nor added new `versionized` structs. Or if I have, these steps must be taken:

>     - [ ] The backwards compatibility tests have been updated and/or new tests added. 

Reviewer: ```suggestion
    - [ ] The backwards compatibility tests have been updated and/or new tests covering the changes ahve been added. 
```

Author: Fixed in a8527285e7721670327f74ca622d2863839b96cc

- [ ] My PR does not contain changes to the critical business logic or cryptographic code. Or if it does then these steps must be taken:
    - [ ] At least two people must be assigned as reviewers (and eventually approve!) the PR.
- [ ] I have not added new structs or modified struct to contain private or key data. Or if so then these steps must be taken:
    - [ ] The `zeroize` and `ZeroizeOnDrop` traits have been implemented to clean up private data.
- [ ] I have not added data to the public storage. Or if I have, then these steps must be taken:
    - [ ] I have ensured that the data does _not_ need to be trusted. I.e. it can be validated through a signature or the existence of a digest in the private storage. 
- [ ] I have not added data to the private storage. Or if I have, then these steps must be taken:

>     - [ ] 

Reviewer: WIP here

Author: Ah sorry. Fixed in a8527285e7721670327f74ca622d2863839b96cc

### Checklist for dependency updates
For dependency updates the following essay questions _must_ be also answered and comments where the import of the dependency happens must be updated if there is any changes in the answers since the last update.
If this is the first time a new dependency is added, then the questions must be answered in the `toml` where the new dependency is imported:
1. Did ownership change change significantly since last update. Is the owner suspicious? I.e. is it limited to one or a few people or small companies in "dangerous territories"?
2. Is the crate not particularly popular?
3. Is there an unusual jump in package versions?
4. Is documentation lacking?
5. Is there no CI enabled on the project's GitHub?

> 6. Do the owners not make any statements in relation security and

Reviewer: ```suggestion
6. Do the owners not make any statements in relation to security and
```
